src/libstd/thread/local.rs

```diff
@@ -364,54 +398,84 @@ pub mod fast {
     impl<T> Key<T> {
         pub const fn new() -> Key<T> {
             Key {
-                inner: UnsafeCell::new(None),
-                dtor_registered: Cell::new(false),
-                dtor_running: Cell::new(false)
+                inner: LazyKeyInner::new(),
+                dtor_state: Cell::new(DtorState::Unregistered),
             }
         }
 
-        pub unsafe fn get(&self) -> Option<&'static UnsafeCell<Option<T>>> {
-            if mem::needs_drop::<T>() && self.dtor_running.get() {
-                return None
+        #[inline]
```

Comment:
Was this tag necessary? With the generics it should already be a candidate for inlining and it seems like LLVM may not need the extra leg-up in recommending it gets inlined.

Reply:
It was not!